src/runtime_src/core/include/xrt/xrt_aie.h

```diff
@@ -163,6 +163,58 @@ class bo : public xrt::bo
   }
 };
 
+class event_impl;
+class event 
+{
+public:
+
+  /**
+   * event() - Constructor from a device
+   *
+   * @param device
+   *  The device on which the profiling event should start
+   *
+   */
+  event(const xrt::device& device);
+
+  /**
+   * start_profiling() - Start AIE performance profiling
+   *
+   * @param option
+   *  Profiling option
+   * @param port1_name
+   *  Profiling port 1 name.
+   * @param port2_name
+   *  Profiling port 2 name.
+   * @param value
+   *  The number of bytes to trigger the profiling event.
+   *
+   * This function configures the performance counters in AI Engine by given
+   * port names and value. The port names and value will have different
+   * meanings on different options.
+   */
+  int 
+  start_profiling(int option, const std::string& port1_name, const std::string& port2_name, uint32_t value) const;
```

Comment:
This will not work in a public API.  What is `option`?  It's an `int` what happens if I pass my lucky number `17`?   Kidding aside, we need a class enum here.   Also, the other parameters are not clear at all, what is profiling port?  What does `number of bytes to trigger the profiling event` mean?  Is 0 bytes good?  It's not useable in a public API without a lot more documentation and explanation as to what happens with out or range numbers.    How do know what port names to use?

Reply:
Noted.
I have added an enum class of profiling option. Also, the other parameters like profiling port etc are the PLIO/ GMIO ports here. I am referring [(UG-1076)](https://docs.xilinx.com/r/en-US/ug1076-ai-engine-environment/Event-Profile-APIs-for-Graph-Inputs-and-Outputs) for implementation.